packages/@aws-cdk/aws-glue/lib/table.ts

```diff
@@ -75,6 +76,38 @@ export enum TableEncryption {
   CLIENT_SIDE_KMS = 'CSE-KMS'
 }
 
+interface BaseStorageParameter {
+  /**
+   * The key of the property. If you want to use a custom key, use `StorageParameters.custom()`.
+   */
+  readonly key: StorageParameters;
+}
+
+/**
+ * A property of the Storage Descriptor.
+ */
+export interface CustomStorageParameter extends BaseStorageParameter {
+  /**
+   * The value of the property.
+   */
+  readonly value: string;
+}
+
+/**
+ * A property of the Storage Descriptor, but with a value provided by the CDK.
+ */
+export interface StorageParameter extends BaseStorageParameter {
+  /**
+   * The value of the property. This is a pre-defined value.
+   */
+  readonly value: ColumnCountMismatchHandlingAction
+  | CompressionType
+  | InvalidCharHandlingAction
+  | NumericOverflowHandlingAction
+  | SurplusBytesHandlingAction
+  | SurplusCharHandlingAction;
+}
```

Comment:
I appreciate the effort here but we can't do this pattern in CDK, since we can't use union types in exported APIs (they become Object for Java customers, which is not a great experience). Instead follow this pattern: 

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-batch/lib/job-definition-base.ts#L175
https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-batch/lib/job-definition-base.ts#L203
https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-batch/lib/job-definition-base.ts#L229

essentially you make `StoageParameter` an enum like class and give it a `custom()` method that in this case will take an arbitrary string.

Reply:
let me know if you need additional clarification on this, I can provide more detail if needed

Reply:
I've created an enum like class called `StorageParameterValue`